pkg/clusteragent/custommetrics/provider.go

```diff
@@ -100,10 +100,16 @@ func (p *datadogProvider) ListAllExternalMetrics() []provider.ExternalMetricInfo
 		extMetric.info = provider.ExternalMetricInfo{
 			Metric: metric.MetricName,
 		}
+		// Avoid overflowing when trying to get a 10^3 precision
+		q, err := resource.ParseQuantity(fmt.Sprintf("%v", metric.Value))
```

Comment:
From my testing `%v` switches between `%f` (classic decimal notation) and `%e` (scientific exponential notation) depending on the value range, which is cool. We'll need to test ParseQuantity works fine with both formats.

Could we add a unit test around ListAllExternalMetrics ensuring several values go through OK?